pkg/labelsfilterdynamic/signals/signal.go

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package signals
+
+type Signal struct {
```

Comment:
It's more heavy-weight, but did you consider using `pkg/trigger` instead of writing up your own logic here?